react-version/src/components/context/BoardContext.js

```diff
@@ -0,0 +1,49 @@
+import React, { createContext, useContext, useState } from "react";
+import { CharaterContext } from "./CharContext";
+import char1 from "../../img/char2.png";
+import char2 from "../../img/char3.png";
+import char3 from "../../img/char4.png";
+import char4 from "../../img/char5.png";
+import char5 from "../../img/char6.png";
+
+const boardData = {
+	btnId: null,
+};
+
+export const BoardContext = createContext();
+
+const Board = ({ children }) => {
+	const [board, setBoard] = useState(boardData);
+	const [char] = useContext(CharaterContext);
+	const charArray = [char1, char2, char3, char4, char5];
+
+	const handleCharacterMove = () => {
+		const destinationHex = document.getElementById(`${board.btnId}`);
+		const characterLink = document.querySelector(`.character`);
+		characterLink.style.display = "block";
+		if (char.selectedChar !== null) {
+			characterLink.style.backgroundImage = `url("${
+				charArray[char.selectedChar]
+			}")`;
+		}
```

Comment:
As we discussed this should be changed to not use `getElementById()` `appendChild()` etc. To remind: character as a separate component -> character image itself may be include in css class, class can be defined dynamically in react.

Reply:
But in this case, every single hex-button must contain div in it. Hex-buttons can't be image containers because character must be able to do animations, that's why it needs separate div. I did character div as a component but I don't know how to move it through clicked buttons. I decided to use appendChild() and make whole mechanic directly in DOM.